packages/core/src/task.ts

```diff
@@ -26,180 +22,221 @@ export interface TaskOptions {
 
 type EnsureHandler = () => void;
 
-export class Task<TOut = unknown> extends EventEmitter implements Promise<TOut>, Trapper {
-  public id = ++COUNTER;
+type WithControls<TOut> = { [CONTROLS]?: Controls<TOut> }
+
+export interface Task<TOut = unknown> extends Promise<TOut> {
+  readonly id: number;
+  readonly state: State;
+  readonly result?: TOut;
+  readonly error?: Error;
```

Comment:
I'm wondering what the use case is for making these part of the public API? Shouldn't these values be consumed via `then()` and `catch()`? There's no way to consume them synchronously, so if I were coming to effection with fresh eyese, I'd be confused as to what they represent.

Reply:
I've moved these to the private `controls` interface.